srcs/app/http/response/delete/HttpDeleteResponse.cpp

```diff
@@ -3,18 +3,64 @@
 HttpDeleteResponse::HttpDeleteResponse(
 	RequestConfig *requestConfig,
 	HttpRequest *request) : HttpBaseResponse(requestConfig, request) {
-	// TODO(any) Implement DELETE
 	if (error_code_) {
-		raw_response_ = HttpErrorResponse(
-										error_code_,
-										request_config_,
-										request_).Content();
+		SetErrorRawResponse_(error_code_);
 	} else {
-		HttpResponse::HeadersMap headers;
-		std::string	body = "Responding to a DELETE request\n";
+		const std::string full_path =
+				request_config_->GetRoot() + request_->GetDecodedPath();
+		try {
+			File file(full_path);
 
-		headers.insert(std::make_pair("Content-Type", "text/plain"));
-		headers.insert(std::make_pair("Location", request_config_->GetReturnUrl()));
-		SetRawResponse_(request_config_->GetReturnStatus(), headers, body);
+			if (file.IsRegularFile() || file.HasEndSlash()) {
+				Delete_(file);
+			} else {
+				SetErrorRawResponse_(409);
+			}
+		} catch(File::Error & e) {
+			SetErrorRawResponse_(e.what());
+		}
 	}
 }
+
+void	HttpDeleteResponse::Delete_(File const &file) {
+	if (file.IsDirectory()) {
+		BuildResponse_(
+			nftw(file.GetPath().c_str(),
+					RemoveSingleFile_,
+					64,
+					FTW_DEPTH | FTW_PHYS));
+	} else {
+		BuildResponse_(
+			RemoveSingleFile_(file.GetPath().c_str(), NULL, 0, NULL));
+	}
+}
+
+int	HttpDeleteResponse::RemoveSingleFile_(
+								const char *fpath,
+								const struct stat *sb,
+								int typeflag,
+								struct FTW *ftwbuf) {
+	// cast to void when function not used as callback
+	if (!sb) (void)sb;
+	if (!typeflag) (void)typeflag;
+	if (!ftwbuf) (void)ftwbuf;
+
+    return remove(fpath);
+}
+
+void	HttpDeleteResponse::BuildResponse_(int return_status) {
+	if (return_status) {
+		errno == EACCES
+			? SetErrorRawResponse_(403)
+			: SetErrorRawResponse_(500);
+	} else {
+		DefaultStatusResponse_(200);
```

Comment:
Yes, I know about it. `204` is a succesful response without body. Since we own this webserver and we are only meant to have `Nginx` as an inspiration I found useful to return a small templated response, may be useful from some scenarios. I am also sure that `Nginx` allows to configure some kind of templated response but, from my point of view, this was a little out of the scope of this feature :cold_sweat: